### PR TITLE
Fix logging

### PR DESCRIPTION
**Pull request by idegtiarov**

### CODE REVIEW

File: edx_adapt/api/resources/course_resources.py [hunk 1]
```diff
@@ -170,7 +170,7 @@ def get(self, course_id):
         try:
             exps = self.repo.get_experiments(course_id)
         except DataException as e:
-            print("--------------------\tDATA EXCEPTION: " + str(e))
+            logger.error("Data exception: {}".format(e))
```
Comment (z4y4ts) [inline edx_adapt/api/resources/course_resources.py]: You can use `logger.exception()` here. It will add exception details for automatically. You just need to make sure to write helpful message.

File: edx_adapt/api/resources/course_resources.py [hunk 2]
```diff
@@ -179,5 +179,5 @@ def post(self, course_id):
         try:
             self.repo.post_experiment(course_id, args['experiment_name'], args['start_time'], args['end_time'])
         except DataException as e:
-            print("--------------------\tDATA EXCEPTION: " + str(e))
+            logger.error("Data exception: {}".format(e))
```
Comment (z4y4ts) [inline edx_adapt/api/resources/course_resources.py]: Here and everywhere below: `logger.exception()`

File: edx_adapt/api/resources/tutor_resources.py [hunk 3]
```diff
@@ -111,15 +114,12 @@ def get(self, course_id, user_id):
 """ Global lock for calling choose_next_problem """
 selector_lock = threading.Lock()
 
-
+""" Run the problem selection sequence (in separate thread) """
```
Comment (z4y4ts) [inline edx_adapt/api/resources/tutor_resources.py]: Why move this outside a function?

File: edx_adapt/api/resources/tutor_resources.py [hunk 4]
```diff
@@ -111,15 +114,12 @@ def get(self, course_id, user_id):
...
-    """
-    Run the problem selection sequence (in separate thread).
-    """
     with selector_lock:
-        """@type selector: SelectInterface
-        @type repo: DataInterface"""
-        # print("--------------------\tSELECTOR LOCK ACQUIRED!")
+        """@type selector: SelectInterface"""
+        """@type repo: DataInterface"""
+        print("--------------------\tSELECTOR LOCK ACQUIRED!")
```
Comment (z4y4ts) [inline edx_adapt/api/resources/tutor_resources.py]: Here and below: `print()` -> `logger.info()`

File: edx_adapt/api/resources/tutor_resources.py [hunk 5]
```diff
@@ -148,10 +148,8 @@ def run_selector(course_id, user_id, selector, repo):
             print("--------------------\tSELECTION NOT REQUIRED!")
 
 
+""" Post a user's response to their current problem """
```
Comment (z4y4ts) [inline edx_adapt/api/resources/tutor_resources.py]: Why this change?

File: edx_adapt/select/skill_separate_random_selector.py [hunk 6]
```diff
@@ -37,9 +38,9 @@ def __init__(self, data_interface, model_interface, parameter_access_mode=""):
         """
         super(SkillSeparateRandomSelector, self).__init__(data_interface, model_interface)
 
-        print self.data_interface
-        print self.model_interface
-        print self.model_interface.get_probability_correct
+        logger.inf(self.data_interface)
```
Comment (z4y4ts) [inline edx_adapt/select/skill_separate_random_selector.py]: Typo.

File: edx_adapt/api/resources/course_resources.py (hunk 1 above)
Comment (idegtiarov) [inline edx_adapt/api/resources/course_resources.py]: Good catch, done

File: edx_adapt/api/resources/course_resources.py (hunk 2 above)
Comment (idegtiarov) [inline edx_adapt/api/resources/course_resources.py]: done

File: edx_adapt/api/resources/tutor_resources.py (hunk 3 above)
Comment (idegtiarov) [inline edx_adapt/api/resources/tutor_resources.py]: It seems that I revert my change, will move doc string back

File: edx_adapt/api/resources/tutor_resources.py (hunk 4 above)
Comment (idegtiarov) [inline edx_adapt/api/resources/tutor_resources.py]: Done

File: edx_adapt/api/resources/tutor_resources.py (hunk 5 above)
Comment (idegtiarov) [inline edx_adapt/api/resources/tutor_resources.py]: It is not change of this  patch it seems that it is result of incomplete merge
